org.scala-ide.sdt.core/src/org/scalaide/core/internal/decorators/markoccurrences/ScalaOccurrencesFinder.scala

```diff
@@ -24,28 +22,6 @@ case class Occurrences(name: String, locations: List[IRegion])
 class ScalaOccurrencesFinder(unit: InteractiveCompilationUnit) extends HasLogger {
   import ScalaOccurrencesFinder._
 
-  /* IMPORTANT:
-   * In the current implementation, all access to `indexCache` are confined to the Presentation
-   * Compiler thread (and this is why `indexCache` requires no synchronization policy).
-   */
-  private var indexCache: Option[TimestampedIndex] = None
-
-  private def getCachedIndex(lastModified: Long, currentCompiler: IScalaPresentationCompiler): Option[MarkOccurrencesIndex] = indexCache match {
-    case Some(TimestampedIndex(`lastModified`, WeakReference(index))) if index.global eq currentCompiler => Some(index)
```

Comment:
How come this test is not enough to ensure consistency? Are you sure this actually fixes the problem you reported?

Also, keep an eye on performance. This cache was added for a reason. May not be a huge problem since the feature can be easily turned off, but then the default should be off.